{"url": "https://api.github.com/repos/rust-lang/rust/issues/80043", "repository_url": "https://api.github.com/repos/rust-lang/rust", "labels_url": "https://api.github.com/repos/rust-lang/rust/issues/80043/labels{/name}", "comments_url": "https://api.github.com/repos/rust-lang/rust/issues/80043/comments", "events_url": "https://api.github.com/repos/rust-lang/rust/issues/80043/events", "html_url": "https://github.com/rust-lang/rust/issues/80043", "id": 767054090, "node_id": "MDU6SXNzdWU3NjcwNTQwOTA=", "number": 80043, "title": "The compiler gets hopelessly confused on `enum S(usize);`", "user": {"login": "jyn514", "id": 23638587, "node_id": "MDQ6VXNlcjIzNjM4NTg3", "avatar_url": "https://avatars.githubusercontent.com/u/23638587?v=4", "gravatar_id": "", "url": "https://api.github.com/users/jyn514", "html_url": "https://github.com/jyn514", "followers_url": "https://api.github.com/users/jyn514/followers", "following_url": "https://api.github.com/users/jyn514/following{/other_user}", "gists_url": "https://api.github.com/users/jyn514/gists{/gist_id}", "starred_url": "https://api.github.com/users/jyn514/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/jyn514/subscriptions", "organizations_url": "https://api.github.com/users/jyn514/orgs", "repos_url": "https://api.github.com/users/jyn514/repos", "events_url": "https://api.github.com/users/jyn514/events{/privacy}", "received_events_url": "https://api.github.com/users/jyn514/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 235791, "node_id": "MDU6TGFiZWwyMzU3OTE=", "url": "https://api.github.com/repos/rust-lang/rust/labels/A-diagnostics", "name": "A-diagnostics", "color": "f7e101", "default": false, "description": "Area: Messages for errors, warnings, and lints"}, {"id": 27424086, "node_id": "MDU6TGFiZWwyNzQyNDA4Ng==", "url": "https://api.github.com/repos/rust-lang/rust/labels/A-parser", "name": "A-parser", "color": "f7e101", "default": false, "description": "Area: The parsing of Rust source code to an AST."}, {"id": 211668100, "node_id": "MDU6TGFiZWwyMTE2NjgxMDA=", "url": "https://api.github.com/repos/rust-lang/rust/labels/T-compiler", "name": "T-compiler", "color": "bfd4f2", "default": false, "description": "Relevant to the compiler team, which will review and decide on the PR/issue."}, {"id": 650731663, "node_id": "MDU6TGFiZWw2NTA3MzE2NjM=", "url": "https://api.github.com/repos/rust-lang/rust/labels/C-bug", "name": "C-bug", "color": "f5f1fd", "default": false, "description": "Category: This is a bug."}, {"id": 1659943986, "node_id": "MDU6TGFiZWwxNjU5OTQzOTg2", "url": "https://api.github.com/repos/rust-lang/rust/labels/D-verbose", "name": "D-verbose", "color": "c9f7a3", "default": false, "description": "Too much output caused by a single piece of incorrect code"}], "state": "open", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 3, "created_at": "2020-12-15T00:40:34Z", "updated_at": "2020-12-16T22:22:23Z", "closed_at": null, "author_association": "MEMBER", "active_lock_reason": null, "body": "I think the error speaks for itself.\r\n\r\nHere is the change I made, applied to 025f0e649e3a9869e42148ee8cf3041e24a13e8c.\r\n\r\n```diff\r\ndiff --git a/src/librustdoc/clean/types.rs b/src/librustdoc/clean/types.rs\r\nindex 26b630440a1..4c2f3b5ce43 100644\r\n--- a/src/librustdoc/clean/types.rs\r\n+++ b/src/librustdoc/clean/types.rs\r\n@@ -1163,33 +1163,7 @@ fn def_id(&self) -> Option<DefId> {\r\n }\r\n \r\n #[derive(Clone, PartialEq, Eq, Hash, Copy, Debug)]\r\n-crate enum PrimitiveType {\r\n-    Isize,\r\n-    I8,\r\n-    I16,\r\n-    I32,\r\n-    I64,\r\n-    I128,\r\n-    Usize,\r\n-    U8,\r\n-    U16,\r\n-    U32,\r\n-    U64,\r\n-    U128,\r\n-    F32,\r\n-    F64,\r\n-    Char,\r\n-    Bool,\r\n-    Str,\r\n-    Slice,\r\n-    Array,\r\n-    Tuple,\r\n-    Unit,\r\n-    RawPointer,\r\n-    Reference,\r\n-    Fn,\r\n-    Never,\r\n-}\r\n+crate enum PrimitiveType(Symbol);\r\n \r\n #[derive(Clone, PartialEq, Eq, Hash, Copy, Debug)]\r\n crate enum TypeKind {\r\n```\r\n[errors.log](https://github.com/rust-lang/rust/files/5692406/errors.log) **edit: this is the wrong logfile, see below**\r\n\r\nI think the main thing making this so awful (1429 errors) is that it continues to try and do name resolution afterwards. I would not expect any errors other than further syntax errors to be reported.", "closed_by": {"login": "jyn514", "id": 23638587, "node_id": "MDQ6VXNlcjIzNjM4NTg3", "avatar_url": "https://avatars.githubusercontent.com/u/23638587?v=4", "gravatar_id": "", "url": "https://api.github.com/users/jyn514", "html_url": "https://github.com/jyn514", "followers_url": "https://api.github.com/users/jyn514/followers", "following_url": "https://api.github.com/users/jyn514/following{/other_user}", "gists_url": "https://api.github.com/users/jyn514/gists{/gist_id}", "starred_url": "https://api.github.com/users/jyn514/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/jyn514/subscriptions", "organizations_url": "https://api.github.com/users/jyn514/orgs", "repos_url": "https://api.github.com/users/jyn514/repos", "events_url": "https://api.github.com/users/jyn514/events{/privacy}", "received_events_url": "https://api.github.com/users/jyn514/received_events", "type": "User", "site_admin": false}, "reactions": {"url": "https://api.github.com/repos/rust-lang/rust/issues/80043/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/rust-lang/rust/issues/80043/timeline", "performed_via_github_app": null, "state_reason": null}